core/src/banking_stage/transaction_scheduler/receive_and_buffer.rs

```diff
@@ -459,6 +460,7 @@ impl TransactionViewReceiveAndBuffer {
                         *result = Err(err);
                         num_dropped_on_status_age_checks += 1;
                         container.remove_by_id(priority_id.id);
+                        continue;
                     }
```

Comment:
all behavior changes here or above (the 2 `continue`s). In original PR (#5180), these were their own commit, and tests were in separate commits. If anyone in backport squad would prefer that structure in this PR, let me know, otherwise I will leave it as is.